src/hooks/useReferralCapture/useReferralCapture.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react'
+import { useLocation } from 'react-router-dom'
+
+const REFERRAL_CODE_KEY = 'shapeshift.referralCode'
+
+/**
+ * Captures referral code from URL and stores it in localStorage
+ * This hook should be called early in the app lifecycle
+ */
```

Comment:
Seems fragile and would potentially override as a non-wanted side-effect. Should we assume that if there's already a referral code in localStorage, shouldn't capture/set anything? 
i.e referral should only happen once